cmd/faucet/faucet.go

```diff
@@ -650,9 +648,10 @@ func (f *faucet) loop() {
 
 		case <-f.update:
 			// Pending requests updated, stream to clients
+			reqData, _ := json.Marshal(map[string]interface{}{"requests": f.reqs})
```

Comment:
If you need the lock for the internal data structs like `f.reqs` than this should also happen after the lock